### PR TITLE
#277 Added Organizations to User API.

### PR DESCRIPTION
**Pull request by criske**
- implemented and tested(where needed) organizations in User implementations.
- moved organizations uri logic from Github to GithubOrganizations.

Left puzzle to start implementing Gitlab organizations (groups).

PR for #277

### CODE REVIEW

File: self-core-impl/src/main/java/com/selfxdsd/core/StoredUser.java [hunk 1]
```diff
@@ -102,6 +112,24 @@ public Projects projects() {
...
 
+    @Override
+    public Organizations organizations() {
+        final Organizations organizations;
+        switch (this.provider){
+            case Provider.Names.GITHUB: organizations =
+                new GithubOrganizations(this.resources,
+                    URI.create("https://api.github.com"), this,
+                    this.storage);
+                break;
```
Comment (criske) [inline self-core-impl/src/main/java/com/selfxdsd/core/StoredUser.java:123]: @amihaiemil Having Provider.Names as enum that encapsulates the name and provider Uri would be really handy :).
If not just class of constants (eg: Provider.Uri) would do it too. Right now Github uri is copy/pasted in a bunch of places.

**Comment by criske** [comment]
@amihaiemil Tests for `BaseSelf.Authenticated`, `PmUser` are missing. Should I write them in this PR or open new issue?

File: self-core-impl/src/main/java/com/selfxdsd/core/StoredUser.java (hunk 1 above)
Comment (amihaiemil) [reply self-core-impl/src/main/java/com/selfxdsd/core/StoredUser.java:123]: @criske I see, but let's leave it as it is for now. We'll make these kind of small imrovements later :D

**Comment by amihaiemil** [comment]
@criske 

> Tests for BaseSelf.Authenticated, PmUser are missing. Should I write them in this PR or open new issue?

Please open another issue :)

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**